src/api/java/mekanism/api/chemical/Chemical.java

```diff
@@ -1,120 +1,251 @@
+    private Map<Class<? extends ChemicalAttribute>, ChemicalAttribute> attributeMap;
+    @Deprecated(forRemoval = true, since = "10.7.11")
+    private double legacyRadioactivity;
+    @Deprecated(forRemoval = true, since = "10.7.11")
+    private boolean hasLegacyAttributesWithValidation;
+    @Nullable
+    @Deprecated(forRemoval = true, since = "10.7.11")
+    private final TagKey<Item> legacyOreTag;
+    @Nullable
+    @Deprecated(forRemoval = true, since = "10.7.11")
+    private TagKey<Item> oreTag;
+    @Deprecated(forRemoval = true, since = "10.7.11")
+    private final boolean isGaseous;
+
+    @SuppressWarnings("removal")
     public Chemical(ChemicalBuilder builder) {
-        //Copy the map to support addAttribute
-        this.attributeMap = new HashMap<>(builder.getAttributeMap());
         this.iconLocation = builder.getTexture();
         this.tint = builder.getTint();
-        this.isRadioactive = attributeMap.containsKey(Radiation.class);
-        this.hasAttributesWithValidation = isRadioactive || attributeMap.values().stream().anyMatch(ChemicalAttribute::needsValidation);
-        this.oreTag = builder.getOreTag();
+        //TODO - 1.22: Remove constructor logic that is below here
+        //Copy the map to support addAttribute
+        this.legacyAttributeMap = new HashMap<>(builder.getAttributeMap());
+        for (ChemicalAttribute legacyAttribute : legacyAttributeMap.values()) {
+            if (legacyAttribute instanceof Radiation radiation) {
+                this.radioactivity = this.legacyRadioactivity = radiation.getRadioactivity();
+            } else if (legacyAttribute.needsValidation()) {
+                //Skip radioactive attributes when checking if we have any that need validation, so that we properly return false
+                // when the radiation manager is disabled
+                this.hasAttributesWithValidation = this.hasLegacyAttributesWithValidation = true;
+            }
+        }
+        this.oreTag = this.legacyOreTag = builder.getOreTag();
         this.isGaseous = builder.isGaseous();
     }
 
     @Override
-    public String toString() {
-        return "[Chemical: " + getRegistryName() + "]";
+    public final String toString() {
+        return MekanismAPI.CHEMICAL_REGISTRY.wrapAsHolder(this).getRegisteredName();
     }
 
     @NotNull
     @Override
-    public Chemical getChemical() {
+    public final Chemical getChemical() {
         return this;
     }
 
     @Override
     public String getTranslationKey() {
         if (translationKey == null) {
-            translationKey = Util.makeDescriptionId("chemical", getRegistryName());
+            //Note: Because chemical registry has a default key, we have to query the name via the resource key so that we can pass null
+            // to makeDescriptionId in cases when our chemical is unregistered
+            translationKey = Util.makeDescriptionId("chemical", MekanismAPI.CHEMICAL_REGISTRY.getResourceKey(this)
```

Comment:
I'd really prefer this to use `getKey` with a default check (or better, pr to Neo adding a nullable `getKey` for `DefaultedMappedRegistry`)

Reply:
We can't directly just have a default check, as we want the empty chemical to have a translatable name. Personally I think we just leave it like this for now, and maybe at some point make a PR to Neo (I can add a TODO in the code about it if you want), or maybe when moving to data pack registries this will just go away to an extent.

Reply:
PR made: https://github.com/neoforged/NeoForge/pull/2013, and I will add a TODO in code so that we don't have to hold merging back on waiting for it, and can just switch to it when we are able to.